colossalai/shardformer/layer/linear.py

```diff
@@ -97,12 +98,22 @@ def __init__(self,
         else:
             assert bias_ is None, 'bias_ must be None if weight is None'
 
+        # pad weight when necessary
+        tp_size = dist.get_world_size(process_group)
+        if out_features % tp_size != 0:
+            self.out_features = tp_size * (out_features // tp_size + 1)
+
         # Parameters.
+        factory_kwargs = {'device': device, 'dtype': dtype}
         if weight is None:
-            factory_kwargs = {'device': device, 'dtype': dtype}
             self.weight = Parameter(torch.empty(self.out_features, self.in_features, **factory_kwargs))
         else:
-            weight.data = weight.data.to(device=device, dtype=dtype)
+            if self.out_features > self.original_out_features:
+                backup_weight_data = weight.data
+                weight.data = Parameter(torch.zeros(self.out_features, self.in_features, **factory_kwargs))
```

Comment:
.data should be a normal tensor instead of parameter

Reply:
I think `torch.nn.functional.pad` is clearer. Why not use this? It current implementation more efficient?